src/detect.c

```diff
@@ -1288,6 +1288,9 @@ static void DetectRunTx(ThreadVars *tv,
     AppLayerGetTxIterState state;
     memset(&state, 0, sizeof(state));
 
+    /* Invariant across transactions */
+    det_ctx->p = p;
```

Comment:
could move this to the general per packet detect entry point

Reply:
Like `DetectRunSetup` and `DetectRunCleanup`?

Reply:
Yeah I think that would make sense.